### PR TITLE
Run unit tests with dotnet test instead of xunit runner

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/659

Regression? Last working version:

## Description
* Run tests under `dotnet test`
* Use TRX format for tests results and update pipeline tasks to upload them
* Get rid of usage of XUnit XML test logger
* Reduce default output verbosity for tests to minimal unless `System.Debug` is `true`
* Reduce `MSBuild.exe` output verbosity to match verbosity of `dotnet build`
* Disable some tests that don't seem to work and I'll work on re-enabling them
https://github.com/NuGet/Client.Engineering/issues/1484

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build/xunit.runsettings [hunk 1]
```diff
@@ -1,7 +1,10 @@
 <RunSettings>
```
Comment (nkolev92) [inline build/xunit.runsettings:1]: Do we still need this?

Comment (jeffkl) [reply build/xunit.runsettings:1]: I think so, its being passed via the command-line to some commands:

https://github.com/NuGet/NuGet.Client/pull/4497/files#diff-22d0028b46aaa349a79daea8ecb058bec0d2b1a6b262bce683c99e5d62d1696cR46

Comment (erdembayar) [reply build/xunit.runsettings:1]: Off topic question: @jeffkl  how did you select line from PR diff?

Comment (jeffkl) [reply build/xunit.runsettings:1]: You can click on the line numbers in the diff to get a link to that part.

Comment (erdembayar) [reply build/xunit.runsettings:1]: Got it.

**Review by nkolev92 — APPROVED** [review]
This looks great.

Any change that removes things from bootstrap.proj makes me super happy :D

File: build/xunit.runsettings (hunk 1 above)
Comment (nkolev92) [reply build/xunit.runsettings:1]: Ok, so looks like these settings are not xunit.runner only, they'd be respect by dotnet test as well.

Comment (jeffkl) [reply build/xunit.runsettings:1]: Yes technically we’re still using the xunit “runner” but it runs under vstest.console instead of xunit.console.

**Pull request closed without merging**